base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
+    if !pto.is_null() {
+        Box::from_raw(pto);
+    }
+}
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------- Compound Inputs, Outputs, Kernels ------------------------- ///
+/// Initialize a Transaction struct to be populated
+
+pub struct TransactionInputs(Vec<TransactionInput>);
+pub struct TransactionOutputs(Vec<TransactionOutput>);
+pub struct TransactionKernels(Vec<TransactionKernel>);
+
+#[no_mangle]
+pub unsafe extern "C" fn transactioninputs_add_transaction_input(
+    inputs: *mut TransactionInputs,
+    transaction: *mut TransactionInput,
 ) -> bool
 {
-    // append this output to PendingTransactionOutputs.outputs_to_be_spent
+    if inputs.is_null() {
+        return false;
+    }
+
+    if transaction.is_null() {
+        return false;
+    }
+
+    (*inputs).0.push((*transaction).clone());
+    return true;
 }
 
-/// Append an UnblindedOutput to be recieved to the pending transaction outputs object
 #[no_mangle]
-pub unsafe extern "C" fn add_output_to_received(
-    pending_tx: *mut PendingTransactionOutputs,
-    value: c_ulonglong,
-    spending_key: *const c_char,
-    feature_flags: c_char, // OutputFlags,
-    maturity: c_ulonglong, // u64
+pub unsafe extern "C" fn transactioninputs_create() -> *mut TransactionInputs {
+    let i = TransactionInputs(Vec::new());
+    Box::into_raw(Box::new(i))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactioninputs_destroy(inputs: *mut TransactionInputs) {
+    if inputs.is_null() {
+        let i = Box::from_raw(inputs);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactionoutputs_create() -> *mut TransactionOutputs {
+    let i = TransactionOutputs(Vec::new());
+    Box::into_raw(Box::new(i))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactionoutputs_destroy(inputs: *mut TransactionOutputs) {
+    if inputs.is_null() {
+        let i = Box::from_raw(inputs);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactionkernels_create() -> *mut TransactionKernels {
+    let i = TransactionKernels(Vec::new());
+    Box::into_raw(Box::new(i))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactionkernels_destroy(kernels: *mut TransactionKernels) {
+    if kernels.is_null() {
+        let i = Box::from_raw(kernels);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn transactioninput_add_transaction_input(
+    inputs: *mut TransactionInputs,
+    transaction: *mut TransactionInput,
 ) -> bool
 {
-    // append this output to PendingTransactionOutputs.outputs_to_be_received
+    if inputs.is_null() {
+        return false;
+    }
+
+    if transaction.is_null() {
+        return false;
+    }
+
+    (*inputs).0.push((*transaction).clone());
+    return true;
 }
 
-/// Add an output to the wallet. `spent` is a boolean that indicates if this output is a spent or unspent output.
+/// Add a transaction output to a transaction struct
```

Comment:
This comment speaks of a `transaction` object but it dealing with a `TransactionOutputs` object.

Reply:
Fixed in next commit, oversight